File: leetcode-algorithms/557. Reverse Words in a String III/557.py
"""
reverse-words-in-a-string-iii
"""
class Solution:
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        return self.v_1(s)

    def v_1(self, s):
        ans = []
        for s in s.split(" "):
            ans.append(s[::-1])
        return " ".join(ans)

    def v_2(self, s):
        return " ".join([i[::-1] for i in s.split(" ")])

File: leetcode-algorithms/557. Reverse Words in a String III/557.js
/**
 * Created by lbp on 2019/3/29.
 */
function v1(s) {
    let array=s.split(" ");
    var result=[]
    for (item of array) {
        result.push(item.split("").reverse().join(""))
    }
    return result.join(" ");
}

// 在v1的基础上利用map函数改进
function v2(s) {
    let array=s.split(" ");
    var result= array.map(item => {
        return item.split("").reverse().join("");
    })
    return result.join(" ");
}
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    return v2(s);
};

console.log(reverseWords("Let's take LeetCode contest"));